sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/FeedResponse.java

```diff
@@ -306,6 +307,23 @@ public String getActivityId() {
         return getValueOrNull(header, HttpConstants.HttpHeaders.ACTIVITY_ID);
     }
 
+    /**
+     * Gets the correlationActivity ID for the responses of a query operation or null if
+     * no correlation activity id is present
+     *
+     * @return the correlation activity id or null if no correlation activity id is present.
+     */
+    public UUID getCorrelationActivityId() {
```

Comment:
I don't think the BE was wired to pass the CorrelaedActivityId back in the response

Reply:
It was - I have a unit test validating it. See https://msdata.visualstudio.com/CosmosDB/_git/CosmosDB?path=/Product/Microsoft.Azure.Documents/SharedFiles/Rntbd/RntbdConstants.cs&version=GBmaster&line=1033&lineEnd=1034&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents https://msdata.visualstudio.com/CosmosDB/_git/CosmosDB?path=/Product/Microsoft.Azure.Documents/SharedFiles/Rntbd/RntbdConstants.cs&version=GBmaster&line=1190&lineEnd=1191&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents